src/lib.rs

```diff
@@ -883,50 +871,39 @@ impl<T> Drop for BufferAsyncMapping<T> {
     }
 }
 
-struct BufferMapReadAsyncUserData<T, F>
+struct BufferMapReadAsyncUserData<F>
 where
-    T: FromBytes,
-    F: FnOnce(BufferMapAsyncResult<&[T]>),
+    F: FnOnce(BufferMapAsyncResult<&[u8]>),
 {
-    size: BufferAddress,
+    size: usize,
```

Comment:
why is this type changed?

Reply:
To get rid of this:

https://github.com/gfx-rs/wgpu-rs/blob/a5921daeb495263b205ebe41aa8a106c3c7fb65e/src/lib.rs#L925

`BufferMapXxxAsyncUserData` is only used to pass info to the C callback in the respective `map_xxx_async`, and the input to those functions is already a `usize`. The current code converts to `BufferAddress` and back to `usize`.